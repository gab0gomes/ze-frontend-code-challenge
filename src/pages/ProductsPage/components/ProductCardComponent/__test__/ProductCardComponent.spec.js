import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import flushPromises from 'flush-promises'

import ProductCard from '../index'

const localVue = createLocalVue()

const product = () => ({
  id: '8877',
  title: 'Brahma 269ml - Unidade',
  rgb: false,
  images: [
    {
      url:
        'https://courier-images-codechallenge.s3-us-west-2.amazonaws.com/product/00008877_02c7ebe9-c23a-405c-8af8-8f0988f276c7.jpg',
      __typename: 'ProductImage'
    }
  ],
  productVariants: [
    {
      availableDate: '2018-10-31T00:00:00',
      productVariantId: '8511',
      price: 19.9,
      inventoryItemId: '80139',
      shortDescription: null,
      title: 'Brahma 269ml - Unidade',
      published: null,
      volume: '0,00269',
      volumeUnit: null,
      description: null,
      subtitle: null,
      components: [],
      __typename: 'ProductVariant'
    }
  ],
  __typename: 'Product'
})

describe('ProductCardComponent', () => {
  it('render correctly', async () => {
    const wrapper = shallowMount(ProductCard, {
      localVue,
      propsData: {
        product: product()
      }
    })
    await flushPromises()

    expect(wrapper.element).toMatchSnapshot()

    wrapper.destroy()
  })

  it('price format works as expected', async () => {
    const wrapper = mount(ProductCard, {
      localVue,
      propsData: {
        product: product()
      }
    })

    await flushPromises()
    expect(wrapper.vm.formattedPrice).toBe('R$ 19,90')

    wrapper.destroy()
  })
})
