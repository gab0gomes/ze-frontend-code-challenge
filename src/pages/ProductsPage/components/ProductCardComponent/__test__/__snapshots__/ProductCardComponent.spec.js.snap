// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCardComponent render correctly 1`] = `
<div
  class="product-card"
>
  <div
    class="product-description"
  >
    <img
      src="https://courier-images-codechallenge.s3-us-west-2.amazonaws.com/product/00008877_02c7ebe9-c23a-405c-8af8-8f0988f276c7.jpg"
    />
     
    <h5>
      Brahma 269ml - Unidade
    </h5>
  </div>
   
  <div
    class="product-shop"
  >
    <p
      class="price"
    >
      R$ 19,90
    </p>
     
    <div
      class="controls"
    >
      <button
        alt="Remover uma unidade do carrinho"
        class="button remove"
      >
        
        -
      
      </button>
       
      <button
        alt="Adicionar uma unidade ao carrinho"
        class="button add"
      >
        
        +
      
      </button>
    </div>
  </div>
</div>
`;
